lib/dpul_collections/solr.ex

```diff
@@ -96,11 +98,35 @@ defmodule DpulCollections.Solr do
   end
 
   @spec add(list(map()), String.t()) :: {:ok, Req.Response.t()} | {:error, Exception.t()}
-  def add(docs, collection \\ read_collection()) do
-    Req.post(
-      update_url(collection),
-      json: docs
-    )
+  def add(docs, collection \\ read_collection())
+
+  def add(docs, collection) when length(docs) > 1 do
+    response =
+      Req.post!(
+        update_url(collection),
+        json: docs
+      )
+
+    if response.status != 200 do
+      Enum.each(docs, fn doc -> add([doc]) end)
```

Comment:
Suggestion: `docs |> Enum.each(&add/1)` or `Enum.each(docs, &add/1)`

Reply:
Had to change the guard clauses a bit, but overall, I think it's better now.